src/sobol_sensitivity.jl

```diff
@@ -0,0 +1,143 @@
+function give_rand_p(p_range,p_fixed=nothing,indices=nothing)
+    if p_fixed == nothing
+        p = [(p_range[j][2] -p_range[j][1])*rand() + p_range[j][1] for j in 1:length(p_range)]
+    else
+        p =  zeros(length(p_range))
+        j = 1
+        for i in 1:length(p_range)
+            if i in indices
+                p[i] = p_fixed[j]
+                j += 1
+            else
+                p[i] = (p_range[i][2] -p_range[i][1])*rand() + p_range[i][1]
+            end
+        end
+    end
+    p
+end
+
+function calc_mean_var(f,p_range,N)
+    y1 = Array(f(give_rand_p(p_range)))
+    y0 = zeros(length(y1))
+    v = zeros(length(y1))
+    if length(size(y1)) != 1
+        y0 = reshape(y0,size(y1)[1],size(y1)[2])
+        v = reshape(v,size(y1)[1],size(y1)[2])
+    end
+    for i in 1:N
+        y1 = Array(f(give_rand_p(p_range)))
+        @. y0 += y1
+        @. v += y1^2
+    end
+    y0 = @. y0/N
+    y0_sq = [i.^2 for i in y0]
+    v = @. v/N - y0_sq
+    y0,v
+end
+
+function first_order_var(f,p_range,N,y0)
+    ys = []
+    for i in 1:length(p_range)
+        y = zeros(length(y0))
+        if length(size(y0)) != 1
+            y = reshape(y,size(y0)[1],size(y0)[2])
+        end
+        for j in 1:N
+            p2 = give_rand_p(p_range)
+            p1 = give_rand_p(p_range,[p2[i]],[i])
+            yer =  Array(f(p1)) .* Array(f(p2))
+            @. y += yer
+        end
+        y = @. y/N - y0^2
+        push!(ys,copy(y))
+    end
+    ys
+end
+
+function second_order_var(f,p_range,N,y0)
+    ys = []
+    for i in 1:length(p_range)
+        for j in i+1:length(p_range)
+            y = zeros(length(y0))
+            if length(size(y0)) != 1
+                y = reshape(y,size(y0)[1],size(y0)[2])
```

Comment:
why is this needed?

Reply:
For the solutions of DiffEqs?

Reply:
if you do `zeros(y0)` then you don't need to reshape. Why did you do `zeros` of the length?